repair_order_group/models/repair_order.py

```diff
@@ -0,0 +1,193 @@
+        "repair.order",
+        compute="_compute_grouped_repair_ids",
+        string="Grouped Repairs",
+        help="All other repairs that belong to the same group.",
+    )
+
+    @api.depends("group_id", "group_id.repair_ids")
+    def _compute_grouped_repair_ids(self):
+        """Compute other repairs in the same group (excluding current repair)."""
+        for repair in self:
+            if not repair.group_id:
+                repair.grouped_repair_ids = False
+                continue
+            repair.grouped_repair_ids = repair.group_id.repair_ids - repair
+
+    def action_add_another_repair(self):
+        """Create a new empty repair linked to the same group.
+
+        If the current repair is not yet assigned to a group, a new
+        `repair.order.group` is automatically created first.
+        The new repair will reuse basic logistics fields such as
+        partner, company, and locations, but not copy any parts or products.
+        """
+        self.ensure_one()
+        if not self.group_id:
+            self.group_id = self.env["repair.order.group"].create({})
+
+        # Create new empty repair record within the same group
+        new_repair = self.env["repair.order"].create(
+            {
+                "group_id": self.group_id.id,
+                "partner_id": self.partner_id.id,
+                "company_id": self.company_id.id,
+                "picking_type_id": self.picking_type_id.id,
+                "location_id": self.location_id.id,
+                "location_dest_id": self.location_dest_id.id,
+            }
+        )
+
+        return {
+            "type": "ir.actions.act_window",
+            "res_model": "repair.order",
+            "res_id": new_repair.id,
+            "view_mode": "form",
+            "target": "current",
+        }
+
+    def write(self, vals):
+        """Propagate partner changes across all repairs in the same group.
+
+        When a user changes the partner on one repair, all other repairs
+        of the same group automatically receive the same partner, ensuring
+        data consistency. Recursion is prevented via the context flag
+        `skip_group_sync`.
+        """
+        res = super().write(vals)
+        if "partner_id" in vals and not self.env.context.get("skip_group_sync"):
+            for rec in self:
+                if rec.group_id:
+                    sib = rec.grouped_repair_ids
+                    if sib:
+                        sib.with_context(skip_group_sync=True).write(
+                            {"partner_id": vals["partner_id"]}
+                        )
+        return res
+
+    def _action_repair_confirm(self):
+        """Confirm repair and cascade confirmation to other group members."""
+        res = super()._action_repair_confirm()
+        if not self.env.context.get("skip_group_confirm"):
+            for rec in self:
+                sib = rec.grouped_repair_ids.filtered(lambda r: r.state == "draft")
```

Comment:
Once again: you must check the source code of the modules you are modifying. 
This is the [original function](https://github.com/odoo/odoo/blob/c6d1aea6f0d3493333f51a2a2c264d0c8d8b0f3d/addons/repair/models/repair.py#L632) that you are calling. If also does the sorting, so you don't need to use 'mapped' here.